### ISSUE TITLE
Fix or remove broken gdb tmp dir deletion hook?

### ISSUE THREAD

**Issue by riftEmber** [chapel-lang/chapel#23554]
#### Background
In [gdb.commands](https://github.com/chapel-lang/chapel/blob/main/compiler/etc/gdb.commands) we currently have a hook intended to delete the tmp dir when the program runs or gdb exits: https://github.com/chapel-lang/chapel/blob/5b88afd050b528495804184f2f2b5696ce11a2eb/compiler/etc/gdb.commands#L1-L27

This has been around [since 2008](https://github.com/chapel-lang/chapel/commit/8cd8e7703044c940996c70a1c7aca98b1892d85f), and is meant to ensure we don't leave tmp dirs lying around when we exit weirdly from gdb.

#### Problem
However, the called function `cleanup_for_exit` does not actually delete the tmp dir, at least since https://github.com/chapel-lang/chapel/pull/22111. The variable `tmpdirname` that lives in its own file [built with special treatment](https://github.com/chapel-lang/chapel/blob/5b88afd050b528495804184f2f2b5696ce11a2eb/compiler/llvm/Makefile#L46-L52) is not even set anywhere; I'm not sure how long this has been the case. This means currently this hook is not accomplishing cleaning up the tmp dir.

#### What to do
We could fix this, or remove it as potentially unneeded. Some reasons which might motivate either route:
- It hasn't been working for a while already.
- Since the changes in tmp-dir handling in https://github.com/chapel-lang/chapel/pull/21944, it might not be needed anymore. I haven't checked on this myself.
  - @mppf 's follow-up PR https://github.com/chapel-lang/chapel/pull/22111 suggests: "Consider using a helper object to track the responsibility of deleting the temp dir (e.g. it could be an RAII pattern with an object representing the temp dir that is moved around)." If it does turn out this is needed, we could replace it with a more robust solution like this suggestion.
- I don't think we have anything testing this hook, nor would it be easy to do so.
- We have no equivalent for `lldb` as far as I can tell.
- @vasslitvinov mentioned considering removing it in https://github.com/chapel-lang/chapel/pull/21269 and may have more reasoning.

**Comment by mppf** [comment]
I'd lean towards removing it, given that it hasn't been working and it seems that nobody has been complaining about that. The main thing that would make me change my mind would be if we have people regularly running `gdb` to debug `chpl` on systems where `/tmp` is not already periodically cleaned up. (I do not do that).

**Comment by vasslitvinov** [comment]
+1 to removing it. 

My /tmp gets cluttered with temp dirs anyways, this does not affect me in any way, I remove them whenever I happen to look there.

**Comment by bradcray** [comment]
I'm not invested in this specific mechanism, but am generally not keen on leaving stuff in a tmp directory if we created it and have the ability to remove it — relying on the tmp directory to clean itself, or for a user like Vass to clean up after us doesn't feel appealing to me.

**Comment by riftEmber** [comment]
I tried removing this hook and starting/quitting gdb while watching for tmp dirs, and found that they get cleaned up without it -- I think via our [signal handlers](https://github.com/chapel-lang/chapel/blob/1b832496ccce9dc925873496673bd9e1624e7746/compiler/util/misc.cpp#L1037-L1058) calling `clean_exit`. I checked they get cleaned up with or without compiler driver, whether exited normally or with Ctrl+C, and after a segfault.

So I will go ahead and remove the hook and supporting code; PR https://github.com/chapel-lang/chapel/pull/25462.
